workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/fragments/group/liferay-learn/fragments/article-related-recipes/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+async function createRecipeSuggestions() {
+	const articleId = document.querySelector(
+		'.article-related-recipes'
+	).dataset.articleId;
+
+	const articleKeywords = await Liferay.Util.fetch(
+		`/o/headless-delivery/v1.0/sites/${Liferay.ThemeDisplay.getSiteGroupId()}/structured-contents/by-key/${articleId}`
+	).then((response) => response.json());
+
+	if (articleKeywords.keywords.length) {
+		createRecipesContainer();
+		const formattedKeywords = formatKeywords(articleKeywords);
+
+		await getRecipeKeywordsSearch(formattedKeywords);
+	}
+}
+
+function formatKeywords(responseKeywords) {
+	try {
+		const keywords = responseKeywords.keywords || [];
+
+		const encodedKeywords = keywords.map((keyword) =>
+			encodeURIComponent("'" + keyword + "'")
+		);
+
+		return encodedKeywords.join('%2C');
+	}
+	catch (error) {
+		console.error('Error processing JSON:', error);
+		return '';
+	}
+}
+
+async function getRecipeKeywordsSearch(articleKeywords) {
+	const response = await Liferay.Util.fetch(
+		`/o/headless-delivery/v1.0/structured-content-folders/34105240/structured-contents?fields=friendlyUrlPath%2Ctitle%2CdatePublished%2Cdescription&search=${articleKeywords}&pageSize=3&sort=datePublished%3Adesc`
```

Comment:
try to use something like:

new URLSearchParams({
				fields: 'id,name,logoURL',
				page: pageIndex + 1,
				pageSize: '20',
				sort: 'name:asc',
			})